agent/pkg/utils/har.go

```diff
@@ -0,0 +1,260 @@
+)
+
+
+// BuildCookies
+// Keep it because we might want cookies in the future
+func BuildCookies(rawCookies []interface{}) []har.Cookie {
+	cookies := make([]har.Cookie, 0, len(rawCookies))
+
+	for _, cookie := range rawCookies {
+		c := cookie.(map[string]interface{})
+		expiresStr := ""
+		if c["expires"] != nil {
+			expiresStr = c["expires"].(string)
+		}
+		expires, _ := time.Parse(time.RFC3339, expiresStr)
+		httpOnly := false
+		if c["httponly"] != nil {
+			httpOnly, _ = strconv.ParseBool(c["httponly"].(string))
+		}
+		secure := false
+		if c["secure"] != nil {
+			secure, _ = strconv.ParseBool(c["secure"].(string))
+		}
+		path := ""
+		if c["path"] != nil {
+			path = c["path"].(string)
+		}
+		domain := ""
+		if c["domain"] != nil {
+			domain = c["domain"].(string)
+		}
+
+		cookies = append(cookies, har.Cookie{
+			Name:        c["name"].(string),
+			Value:       c["value"].(string),
+			Path:        path,
+			Domain:      domain,
+			HTTPOnly:    httpOnly,
+			Secure:      secure,
+			Expires:     expires,
+			Expires8601: expiresStr,
+		})
+	}
+
+	return cookies
+}
+
+func BuildHeaders(rawHeaders []interface{}) ([]har.Header, string, string, string, string, string) {
+	var host, scheme, authority, path, status string
+	headers := make([]har.Header, 0, len(rawHeaders))
+
+	for _, header := range rawHeaders {
+		h := header.(map[string]interface{})
+
+		headers = append(headers, har.Header{
+			Name:  h["name"].(string),
+			Value: h["value"].(string),
+		})
+
+		if h["name"] == "Host" {
+			host = h["value"].(string)
+		}
+		if h["name"] == ":authority" {
+			authority = h["value"].(string)
+		}
+		if h["name"] == ":scheme" {
+			scheme = h["value"].(string)
+		}
+		if h["name"] == ":path" {
+			path = h["value"].(string)
+		}
+		if h["name"] == ":status" {
+			path = h["value"].(string)
+		}
+	}
+
+	return headers, host, scheme, authority, path, status
+}
+
+func BuildPostParams(rawParams []interface{}) []har.Param {
+	params := make([]har.Param, 0, len(rawParams))
+	for _, param := range rawParams {
+		p := param.(map[string]interface{})
+		name := ""
+		if p["name"] != nil {
+			name = p["name"].(string)
+		}
+		value := ""
+		if p["value"] != nil {
+			value = p["value"].(string)
+		}
+		fileName := ""
+		if p["fileName"] != nil {
+			fileName = p["fileName"].(string)
+		}
+		contentType := ""
+		if p["contentType"] != nil {
+			contentType = p["contentType"].(string)
+		}
+
+		params = append(params, har.Param{
+			Name:        name,
+			Value:       value,
+			Filename:    fileName,
+			ContentType: contentType,
+		})
+	}
+
+	return params
+}
+
+func NewRequest(request *api.GenericMessage) (harRequest *har.Request, err error) {
+	reqDetails := request.Payload.(map[string]interface{})["details"].(map[string]interface{})
+
+	headers, host, scheme, authority, path, _ := BuildHeaders(reqDetails["headers"].([]interface{}))
+	cookies := make([]har.Cookie, 0) // BuildCookies(reqDetails["cookies"].([]interface{}))
```

Comment:
in the end we probably want to take cookies too and it will be used with this function.
but for now it is not very interesting because we are not really use it as far as I know